sim/src/simulator.rs

```diff
@@ -131,15 +133,15 @@ impl Simulator {
         }
         for i in 0..self.elements.keys().count() {
             let src = &indices_to_sim_el[&i];
-            for (rpc, dst) in &self.rpc_buffer[src] {
+            while !&self.rpc_buffer[src].is_empty() {
+                let (rpc, dst) = &self.rpc_buffer.get_mut(src).unwrap().pop().unwrap();
```

Comment:
This a pretty complex unwrapping and accessing operation. Is this the right way to do this?

Reply:
I believe so.  The rpc buffer is a hashmap from the ids to a vector of rpcs, destinations.  So for each element, we are emptying the element's rpc buffer, and sending those rpcs to the corresponding destinations.  We have to do get_mut because we need to be able to delete rpcs from the buffer, and we have to do pop because we want to consume the rpcs.  Both those functions come with options, so it winds up being long for the unwraps.

Reply:
Okay, I will take a closer look at this later.